src/data_provider/src/extended_sources/users/src/logged_in_users_win.cpp

```diff
@@ -0,0 +1,282 @@
+#include <iostream>
+#include <ctime>
+
+#include "logged_in_users_win.hpp"
+#include "winapi_wrappers.hpp"
+#include "encodingWindowsHelper.h"
+
+
+const std::map<int, std::string> LoggedInUsersProvider::m_kSessionStates =
+{
+    {WTSActive, "active"},
+    {WTSDisconnected, "disconnected"},
+    {WTSConnected, "connected"},
+    {WTSConnectQuery, "connectquery"},
+    {WTSShadow, "shadow"},
+    {WTSIdle, "idle"},
+    {WTSListen, "listen"},
+    {WTSReset, "reset"},
+    {WTSDown, "down"},
+    {WTSInit, "init"}
+};
+
+LoggedInUsersProvider::LoggedInUsersProvider(std::shared_ptr<ITWSapiWrapper> twsWrapper,
+                                             std::shared_ptr<IWinBaseApiWrapper> winBaseWrapper, std::shared_ptr<IWinSDDLWrapper> winSddlWrapper,
+                                             std::shared_ptr<IWinSecurityBaseApiWrapper> winSecurityWrapper)
+    : m_twsApiWrapper(std::move(twsWrapper)),
+      m_winBaseWrapper(std::move(winBaseWrapper)),
+      m_winSddlWrapper(std::move(winSddlWrapper)),
+      m_winSecurityWrapper(std::move(winSecurityWrapper)) {}
+
+LoggedInUsersProvider::LoggedInUsersProvider()
+    : m_twsApiWrapper(std::make_shared<TWSapiWrapper>()),
+      m_winBaseWrapper(std::make_shared<WinBaseApiWrapper>()),
+      m_winSddlWrapper(std::make_shared<WinSDDLWrapper>()),
+      m_winSecurityWrapper(std::make_shared<WinSecurityBaseApiWrapper>()) {}
+
+nlohmann::json LoggedInUsersProvider::collect()
```

Comment:
perhaps the collect function could be separated to make it clearer, for example.

``` c++
std::vector<SessionInfo> LoggedInUsersProvider::enumerateSessions();
SessionInfoDetails LoggedInUsersProvider::buildSessionDetails(const SessionInfo& session);
nlohmann::json LoggedInUsersProvider::serializeSessionDetails(const SessionInfoDetails& details);

```

```c++
nlohmann::json LoggedInUsersProvider::collect() {
    nlohmann::json results;

    for (const auto& session : enumerateSessions()) {
        auto sessionDetails = buildSessionDetails(session);
        results.push_back(serializeSessionDetails(sessionDetails));
    }

    return results;
}

```